core/templates/dev/head/pages/exploration_editor/simple_editor_tab/ScrollSyncDirective.js

```diff
@@ -0,0 +1,45 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive used to mark parts of the editor that have a sidebar
+ * item associated to it, see ScrollSyncService.js.
+ */
+
+oppia.directive('scrollSync', ['ScrollSyncService', 'QuestionIdService',
+  function(ScrollSyncService, QuestionIdService) {
+  return {
+    restrict: 'A',
+    link: function(scope, element, attrs) {
+      if (!(element.prop('tagName') === 'H3')) {
+        console.log(element.tag, attrs.scrollSync);
+        throw Error('ScrollSync is only used with h3 elements.');
+      };
+      // Set the element Id.
+      var subfieldLabels = [
+        'Multiple choice', 'Correct answer', 'Hints', 'Bridge text'];
+      var label = attrs.scrollSync;
+      var id;
+      if (!(subfieldLabels.indexOf(label) === -1)) {
+        var questionId = scope.questionId || scope.question.getId();
```

Comment:
Let's do the latter; it has a better separation of responsibilities. getSubfieldId() can delegate to the common id service, and I think that's fine.